paper-api/src/main/java/org/bukkit/entity/HumanEntity.java

```diff
@@ -701,11 +703,55 @@ default void openSign(@NotNull org.bukkit.block.Sign sign) {
      * This will force the entity to drop the item they are holding with
      * an option to drop the entire {@link ItemStack} or just 1 of the items.
      *
+     * @deprecated You should instead use {@link #dropItem(EquipmentSlot, int)} with a {@link EquipmentSlot#HAND} parameter.
      * @param dropAll True to drop entire stack, false to drop 1 of the stack
      * @return True if item was dropped successfully
      */
+    @Deprecated
     public boolean dropItem(boolean dropAll);
 
+    /**
+     * Makes the entity drop an item from their inventory based on the slot.
+     *
+     * @param slot   The slot to drop
+     * @param amount The number of items to drop from this slot. Values below 1 don't drop an Item
+     * @return The dropped item entity, or null if the action was unsuccessful
+     * @throws IndexOutOfBoundsException If the slot is negative or bigger than the player's inventory
```

Comment:
I think I'd prefer just a standard IllegalArgumentException for an "invalid slot". That is more flexible for the future if slot numbers are ever non-contiguous.